components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -63,8 +70,24 @@ func ignoreNotFound(err error) error {
 // NotebookReconciler reconciles a Notebook object
 type NotebookReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	kubeClient    clientset.Interface
+	Log           logr.Logger
+	Scheme        *runtime.Scheme
+	eventRecorder record.EventRecorder
```

Comment:
I think you can get recorder from manager
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/manager/manager.go#L86